app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -130,6 +145,30 @@ export interface Cognito {
         email: string,
         code: string
     ) => Promise<results.Result<null, ConfirmSignUpError>>
+    /** Signs in via the Google federated identity provider.
+     *
+     * This function will open the Google authentication page in the user's browser. The user will
+     * be asked to log in to their Google account, and then to grant access to the application.
+     * After the user has granted access, the browser will be redirected to the application. */
+    signInWithGoogle: () => Promise<null>
```

Comment:
consider using `Promise<void>` (value of type `void` = this value shouldn't be used) to avoid `.then(() => null)`